### ISSUE TITLE
Unable to dereference schema with $id 'session'

### ISSUE THREAD

**Issue by beeman** [elysiajs/elysia#981]
### What version of Elysia is running?

`latest`

### What platform is your computer?

`MacOS 15.2 (24C101)`

### What steps can reproduce the bug?

Follow the [tutorial](https://elysiajs.com/tutorial#reference-model) where the `optionalSession` part gets explained.

It shows you to create `optionalSession` as an optional reference to the `session` model:

```ts
// repro of the issue without any of the tutorial context
import { Elysia, t } from "elysia";

const service = new Elysia()
  .model({
      session: t.Cookie({token: t.Number()}),
      optionalSession: t.Optional(t.Ref('session')),
    }
  )

const app = new Elysia()
  .use(service)
  .get("/", () => "Hello Elysia", {cookie: 'optionalSession',})
  .listen(3030);

console.log(`🦊 Elysia is running at ${app.server?.hostname}:${app.server?.port}`);
```

### What is the expected behavior?

Reference should work

### What do you see instead?

It throws an error: `Unable to dereference schema with $id 'session'`

### Additional information

I worked around it by referencing the same object twice:

```ts
const cookieModel = t.Cookie({ token: t.Number() }, { secrets: 'seia' })
//
  .model({
    session: cookieModel,
    optionalSession: t.Optional(cookieModel),
  })

```

### Have you try removing the `node_modules` and `bun.lockb` and try again yet?

Yes

**Comment by SaltyAom** [comment]
Should have been fixed with [57b24c1](https://github.com/elysiajs/elysia/commit/57b24c1cf531844607fd0e46f7b6b7a920f117f6)

**Comment by SaltyAom** [comment]
Fixed, and published under [1.2.7](https://github.com/elysiajs/elysia/releases/tag/1.2.7)
